Content.Server/Research/Systems/ResearchSystem.Console.cs

```diff
@@ -97,6 +97,7 @@ private void OnConsoleRegistrationChanged(EntityUid uid, ResearchConsoleComponen
 
     private void OnConsoleDatabaseModified(EntityUid uid, ResearchConsoleComponent component, ref TechnologyDatabaseModifiedEvent args)
     {
+        SyncClientWithServer(uid, raiseEvents: false);
```

Comment:
This kind of boolean is the start of the path to unmaintainable spaghetti.
Why not extract the portions of `SyncClientWithServer` that you do want to call into a new function and call that from here and reuse it in `SyncClientWithServer`?
Basically, one function would be "just synchronize the DB components" and the other would be "synchronize the DB components and raise `TechnologyDatabaseModifiedEvent`"?

Reply:
Maybe the real issue here is that `TechnologyDatabaseModifiedEvent` is raised on the console _even though the local DB copy hasn't actually changed_, so a more correct solution would be to add a new event to discriminate between "The server I sync with changed" and "my internal DB copy changed"?